src/fastnetmon.cpp

```diff
@@ -3651,21 +3651,31 @@ void produce_dpi_dump_for_pcap_dump(std::string pcap_file_path, std::stringstrea
 
         ss << parsed_packet_as_string << " protocol: " << protocol_name << " master_protocol: " << master_protocol_name << "\n";
 
-        // Free up all memory
-        ndpi_free_flow(flow);
-        free(dst);
-        free(src);
-        
-        close(filedesc);
-
         total_packets_number++;
     }
 
+    // Free up all memory
+    ndpi_free_flow(flow);
+    free(dst);
+    free(src);
+    
+    close(filedesc);
+
     amplification_attack_type_t attack_type;
 
     // Attack type in unknown by default
     attack_type = AMPLIFICATION_ATTACK_UNKNOWN;
 
+    char buff[256];
+                                          
+    snprintf(&buff[0],sizeof(buff)-1,"DPI pkt stats: total:%llu DNS:%llu NTP:%llu SSDP:%llu SNMP:%llu",
```

Comment:
Could you replace this code by C++ version of sprintf? You could use something like: 
```
logger << log4cpp::Priority::INFO << "DNS " << dns_amplification_packets ....
```